File: Backend/src/Application/Manutencoes/GetStatus/VeiculoStatusManutencaoResponse.cs
using Domain.Common;

namespace Application.Manutencoes.GetStatus;

public class VeiculoStatusManutencaoResponse
{
    public NivelAlertaEnum StatusGeral { get; set; }
    public int QuantidadeTotal { get; set; }
    public int QuantidadeEmDia { get; set; }
    public int QuantidadeProximas { get; set; }
    public int QuantidadeVencidas { get; set; }
    public List<ManutencaoPendenciaResponse> Pendencias { get; set; }

    public VeiculoStatusManutencaoResponse(
        List<ManutencaoPendenciaResponse> todas,
        List<ManutencaoPendenciaResponse> pendencias)
    {
        Pendencias = pendencias;
        QuantidadeTotal = todas.Count;
        QuantidadeVencidas = todas.Count(p => p.Status == NivelAlertaEnum.Critico);
        QuantidadeProximas = todas.Count(p => p.Status == NivelAlertaEnum.Atencao);
        QuantidadeEmDia = QuantidadeTotal - QuantidadeVencidas - QuantidadeProximas;

        StatusGeral = QuantidadeVencidas > 0
            ? NivelAlertaEnum.Critico
            : QuantidadeProximas > 0
                ? NivelAlertaEnum.Atencao
                : NivelAlertaEnum.Normal;
    }
}

File: Frontend/src/screens/manutencao/api/types/VeiculoStatusManutencaoResponse.ts
import type { NivelAlertaEnum } from "@/screens/common/api/types/NivelAlertaEnum";

import type { ManutencaoPendenciaResponse } from "./ManutencaoPendenciaResponse";

export interface VeiculoStatusManutencaoResponse {
  statusGeral: NivelAlertaEnum;
  quantidadeTotal: number;
  quantidadeEmDia: number;
  quantidadeProximas: number;
  quantidadeVencidas: number;
  pendencias: ManutencaoPendenciaResponse[];
}
